### PR TITLE
Updated create logic to ignore SMD unless in sunrise

### PR DESCRIPTION
**Pull request by Falkenfighter**
Changes in this PR are a reflection of the conversation in this thread: https://groups.google.com/forum/#!topic/nomulus-discuss/3jtsbtd0yU8

This PR address an issue where the existing logic assumed the presence of a SMD means the current flow is a sunrise/sunrush request.

### CODE REVIEW

File: javatests/google/registry/flows/domain/DomainCreateFlowTest.java [hunk 1]
```diff
@@ -253,9 +253,13 @@ private void assertSuccessfulCreate(String domainTld, boolean isAnchorTenant) th
   }
 
   private void assertNoLordn() throws Exception {
+    assertNoLordn(null, null);
+  }
+
+  private void assertNoLordn(String smdId, LaunchNotice launchNotice) throws Exception {
```
Comment (CydeWeys) [inline javatests/google/registry/flows/domain/DomainCreateFlowTest.java]: Tag both parameters with @Nullable.

**Comment by CydeWeys** [comment]
This is going through our internal review process now.

**Comment by Falkenfighter** [comment]
@CydeWeys Any update on this?

**Comment by nfelt** [comment]
Sorry, I had some comments internally that diverged a little from Corey's recommendations (not your fault at all) so we were sorting those out.  We're just going to go ahead and make the tweaks to the internal commit since we use that internal commit process anyway (versus merging), so what you see in the next sync will look a little different from this.

FYI, the main issue is that once GA registrations allow signed marks, doing verifySignedMarksAllowed() doesn't make much sense any more, because then the only case where DomainCreateFlow disallows signed marks is for non-anchor-tenant users before GA, but DomainCreateFlow can't even be invoked by those users before GA (except if they're superuser, but that's a very minor edge case).  So we're just removing that verification instead.

Also, now that hasSignedMarks and isInSunrise are independent variables, we need to be careful about which one we use where - in particular, verifyNotReserved() should be checking isInSunrise (since that boolean value is used to allow special kinds of reserved names that are allowed in sunrise), rather than hasSignedMarks, otherwise GA domain creates with signed marks will be allowed to register names that are supposed to be reserved except for sunrise.

**Comment by CydeWeys** [comment]
It's merged now.

**Comment by Falkenfighter** [comment]
Awesome thanks guys!

**Pull request closed without merging**